resource/cmake/CommonFunctions.cmake

```
@@ -0,0 +1,487 @@
+###############################################################################
+#  (c) 2021 Copyright, Real-Time Innovations, Inc. (RTI) All rights reserved. #
+#                                                                             #
+#  RTI grants Licensee a license to use, modify, compile, and create          #
+#  derivative works of the software solely for use with RTI Connext DDS.      #
+#  Licensee may redistribute copies of the software provided that all such    #
+#  copies are subject to this license.                                        #
+#  The software is provided "as is", with no warranty of any type, including  #
+#  any warranty for fitness for any purpose. RTI is under no obligation to    #
+#  maintain or support the software.  RTI shall not be liable for any         #
+#  incidental or consequential damages arising out of the use or inability to #
+#  use the software.                                                          #
+#                                                                             #
+###############################################################################
+
+#[[
+rtigw_list_prefix
+-----------------
+
+ * Brief: prepend a prefix to all the elements of a list
+ * Params:
+ ** LIST: list to add the prefix to all its elements
+ ** PREFIX: prefix to add
+ ** _OUT_VARIABLE: output list that will contain the list with the prefix
+
+ * How to use it:
+
+    rtigw_list_prefix(MY_OUT_VAR
+        INPUT_LIST my_list
+        PREFIX "${CMAKE_CURRENT_SOURCE_DIR}/"
+    )
+]]
+
+function(rtigw_list_prefix _OUT_VARIABLE)
+    set(_BOOLEANS)
+    set(_SINGLE_VALUE_ARGS INPUT_LIST PREFIX)
+    set(_MULTI_VALUE_ARGS)
+
+    cmake_parse_arguments(_args
+        "${_BOOLEANS}"
+        "${_SINGLE_VALUE_ARGS}"
+        "${_MULTI_VALUE_ARGS}"
+        ${ARGN}
+    )
+
+    set(_result)
+    foreach(_val ${_args_INPUT_LIST})
+        list(APPEND _result "${_args_PREFIX}${_val}")
+    endforeach()
+    set(${_OUT_VARIABLE} ${_result} PARENT_SCOPE)
+endfunction()
+
+function(rtigw_copy_files)
+    set(_BOOLEANS)
+    set(_SINGLE_VALUE_ARGS CUSTOM_TARGET_NAME OUTPUT_DIR)
+    set(_MULTI_VALUE_ARGS LIST_OF_FILES_RELATIVE_PATH)
```

Argument name "simplification":

```suggestion
function(rtigw_copy_files)
    set(_BOOLEANS)
    set(_SINGLE_VALUE_ARGS TARGET OUTPUT)
    set(_MULTI_VALUE_ARGS FILES)
```